pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -457,3 +457,16 @@ def test_to_frame_datetime_tz(self):
         result = idx.to_frame()
         expected = DataFrame(idx, index=idx)
         tm.assert_frame_equal(result, expected)
+
+    def test_index_map(self):
+        # see GH20990
+        count = 6
+        index = pd.date_range("2018-01-01", periods=count, freq="M", name="name")
+        a = pd.Series(np.arange(count), index=index)
+        a.index = a.index.map(lambda x: (x.year, x.month))
+
+        df = pd.DataFrame([[2018, i + 1] for i in range(6)], columns=["name", "name"])
```

Comment:
this is a very complicated expected, can you just construct the multi-index directly